### PR TITLE
Removed _get_original from SinonStub

### PR DESCRIPTION
**Pull request by jonathan-benn**
* Removed recursion and simplified readability
  * SinonStubCondition now starts with
    underscore ("_") to show that it is package
    private and not for end-user consumption
  * Renamed __copy_of to _copy (less awkward).
    Also, after removing _get_original it was
    no longer possible for __copy_of to remain
    private.

### CODE REVIEW

File: sinon/lib/stub.py [hunk 1]
```diff
@@ -133,7 +129,7 @@ def exception_function(*args, **kwargs):
         super(SinonStub, self).wrap2stub(exception_function)
         return self
 
-class SinonStubCondition(SinonStub):
+class _SinonStubCondition(SinonStub):
```
Comment (note35) [inline sinon/lib/stub.py:132]: Interesting naming, I will keep this since it may be replaced by FSM table.

File: sinon/lib/stub.py [hunk 2]
```diff
@@ -147,25 +143,19 @@ def __new__(cls, *args, **kwargs):
         """
         return object.__new__(cls)
 
-    def __init__(self, copy_of, cond_args, cond_kwargs, oncall):
+    def __init__(self, copy, cond_args, cond_kwargs, oncall):
```
Comment (note35) [inline sinon/lib/stub.py:146]: Note: there is a popular library which is also called "copy".

**Comment by note35** [comment]
Note: those repeated code could be refactored by adding two functions for handling all of them. We can do that later.

**Review by note35 — APPROVED** [review]
LGTM. Please take a look at my feedback.

**Comment by jonathan-benn** [comment]
Hi Kir,

Kir wrote:
> Note: those repeated code could be refactored by adding two functions for handling all of them. We can do that later.

Sorry, which repeated code are you talking about? :)

Thanks,

**Comment by note35** [comment]
It's not a big task at this moment,
In [L63-64](https://github.com/note35/sinon/blob/master/sinon/lib/stub.py#L63-L64) and [L87-88](https://github.com/note35/sinon/blob/master/sinon/lib/stub.py#L87-L88) -> make one function
In [L167-169](https://github.com/note35/sinon/blob/master/sinon/lib/stub.py#L167-L169) and [L181-183](https://github.com/note35/sinon/blob/master/sinon/lib/stub.py#L181-L183) -> make the other function

The benefit is not big at this moment, but it might be helpful if the logic become complicated or more new functions are added used this 2 functions.

**Comment by jonathan-benn** [comment]
Hi Kir,

Yes, you are right about the code duplication. I'll add an issue

--Jonathan

**Pull request merged**